src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -890,18 +911,31 @@ impl Step for Rustc {
         });
     }
 
-    /// Builds the compiler.
+    /// Builds the requested compiler at the requested stage.
     ///
-    /// This will build the compiler for a particular stage of the build using
-    /// the `compiler` targeting the `target` architecture. The artifacts
-    /// created will also be linked into the sysroot directory.
+    /// This will build the compiler for a particular stage of the build using the `compiler`
+    /// targeting the `target` architecture. The artifacts created will also be linked into the
+    /// sysroot directory.
+    #[cfg_attr(
+        feature = "tracing",
+        instrument(
+            level = "trace",
+            name = "Rustc::run",
+            skip_all,
+            fields(target = ?self.target, compiler = ?self.compiler),
+        ),
+    )]
     fn run(self, builder: &Builder<'_>) -> u32 {
         let compiler = self.compiler;
```

Comment:
On `Rustc` there's a comment on `self.compiler` that says

```rs
    /// The **previous** compiler used to compile this compiler.
    pub compiler: Compiler,
```

I'm not 100% sure if that's actually true under all scenarios.